src/array.jl

```diff
@@ -500,9 +489,10 @@ function vcat(A::CategoricalArray...)
 
     T = Base.promote_eltype(A...) >: Null ?
         Union{eltype(newlevels), Null} : eltype(newlevels)
-    refs = DefaultRefType[refsvec...;]
+    R = reftype(T)
```

Comment:
AFAICT `T` cannot be a categorical value here, so no need to call `reftype`: just keep using `DefaultRefType`.